[READY] New Side Antagonist: The Creep!

tralezab opened the pull request:
:cl: 
add: Added a new antagonist, the Creep!
add: Chosen from a random event, the Creep will become obsessed with one person, feeling amazing around them and terrible when they aren't. They will have objectives to steal heirlooms, take pictures, hug, and kill coworkers. They also have to kill the obsession but some objectives can only be completed while the obsession is alive, requiring you to protect the obsession!
/:cl:

There really is no reason to betray people, generally. Maybe some stuff with security and traitors working things out, but a traitor based around betrayal and the general insane creep idea is just cool.

One Creep can show up, and they get objectives to mess with their obsession leading to the obsession's death. Here is their objectives:

2. A random creepy objective. It could be taking a photo with them in it, hugging them, etc
3. After your are done with your objectives, kill them. you can't complete the objectives when they are dead!

Creeps do not get the determination mood. Instead, when they are around their obsession, they feel AMAZING! If they are around too long, they will stutter, cough, and generally spaghetti out when talking near their obsession. So if you are going to talk near them, do it when you aren't entranced. 

todo:
- [x] finish up the polaroid obsession
- [x] find a way to start the next timer when the current obsession dies
- [x] now code that way to start the timer (status effects or components)
- [x] some more creep objectives between protecting and killing
- [x] rework this into a side antag

tralezab committed:
most of the work done, so far.

obscolene commented:
can it be like mine craft creepar where it go SSSsss BOOOOMM!!!!! :D

tralezab commented:
Ok very cool

nicbn commented:
Put [wip] in PR name

elyina commented:
I'm not sure if encouraging creepy behavior is a good thing

Ratvar commented:
that moment when ur targets commits roundstart suicide to roll antag

MrDoomBringer commented:
Should have to take a pic of the obsession while the obsession is alive, to promote actually protecting them

tralezab commented:
@elyina fair, dunno how i'd circumvent that though. i could go with the more indicative "two faced creep" which would be more like a creep in the sense that they betray you. thoughts?
@Ratvar what do you mean, if you can suicide to get this you could suicide to get traitor, this doesn't convert or work any way different than traitor (or it shouldn't)
@MrDoomBringer i can do that, it'd be a pretty big giveaway that they are the creep if someone comes up to you and takes a picture of themselves with you in it, so they might do it when they need to kill you. !creepy!

GuyonBroadway commented:
This kinda reminds me of how Micheal Meyers plays in Dead by Daylight which is kinda neat.

Bonus points if the creep can get a knife that does a shittone of damage to the object of obsession or something.

tralezab committed:
polaroid objective complete, groundwork for the deathrattle status effect and the special assassinate

HomocidlePancakes commented:
_Yandere intensifies_

HomocidlePancakes commented:
Here is the official theme of this pull request
https://www.youtube.com/watch?v=KO-G5DVNlw4

anconfuzedrock commented:
I think the objectives might be better with some variety, maybe the time you need to protect could vary, maybe you might need something else instead of a pic (like to eat their liver when you killed them or somethin), maybe the final objective just needs to be protected the whole round so you never know. I’ll never know if this is good advice.

Kierany9 commented:
>I like the idea of making the protect objective something you can't 100% trust

Hold up.

Traitors don't even get a protect objective anymore, only space ninjas, valentines(event), wizard apprentices(protect your master) and traitor AIs(protect _and maroon_) do.

Qustinnus commented:
@elyina
lol who the fuck cares. It's not like someone will actually be a creep because of this antag role. It's like saying we should remove nuke ops so we don't encourage terroristic behavior

Cyberboss reviewed (changes requested):
Suicide shouldn't prevent greentext in this scenario

zxaber commented:
Suicides prevent greentext for hiveminds. There was a round yesterday where all hiveminds had the same target for "protect and assimilate", and the person suicided and redtexted them all.

MrDoomBringer commented:
@tralezab ratvar means that my target could be someone who just joined to roll antag, and would kill themselves in the first few minutes of the round after failing that roll.

Nabski89 commented:
Please don't tie central command officials into this in any way.

tralezab commented:
@MrDoomBringer i'll fix that by not failing the creep, and do you one better. if the first target suicides i can make it add another through my *very robust* creepiness system. 

@Nabski89 sure i just need a reason to add creeps 😅

Militaires commented:
INB4 PEOPLE CRY ABOUT "LE CREEPY BEHAVIOUR REEE"

tralezab commented:
i must have done this too soon to the neet pr huh, getting some aftershock in here

also new objective idea: escaping with some of their skin. you can't get the skin unless you find a gibber, which makes the objective kind of hard to do. nice

tralezab commented:
the real question is: should I have the in love effect for the creep? it would be cool but i do kinda want this merged hmm

k-k-karma commented:
hello mr myers

Swiftkor commented:
Space serial killer, sounds great. My only issue is now I’m going to be valid any time I just want to take a picture. Also shitsec is going to start executing people for possession of a camera.

GuyonBroadway commented:
If you didn't want creepy behaviour in ss13 you'd get rid of every non-human race in the game starting with catgirls for good, then closely followed by the lizards.


Also the creepster should have more objectives besides taking a photo to avoid the aforementioned "camera = VALID" issue.

tralezab commented:
^^ working on this right now. Bonus objectives to kill people in certain places, skinning, their heirlooms becoming yours, and more.

not even touching the first part though that's how you get 91 comments for a shower pr

yoyobatty commented on code/datums/status_effects/neutral.dm:
```diff
@@ -69,3 +69,11 @@
+	status_type = STATUS_EFFECT_UNIQUE
+	var/datum/antagonist/antagonist
+
+/datum/status_effect/deathcheck
```
duplicate line ?

tralezab replied:
if you are reviewing this you should get all the new shit here one second

yoyobatty commented:
Lock someone in cell with no trap door and escape with plasma behind glass door, set timer, door opens and kills them.

tralezab committed:
description for changes

new assassinate with deathrattle check,, repeating objectives system

ExcessiveUseOfCobblestone commented:
This should be a side antag tbh, your feelings for someone form into this crazy obsession. It also gives you a position where you could be in contact with the person not with the intention of an objective and then become this weirdo, really immersive imo (not speaking from experience outside of hollywood films).

tralezab commented:
Why not both? We could have a roundstart creep that has a list of obsessions they go through, with each obsession getting more and more difficult objectives to do before they murder them and move on.

And we can have the side antagonist creep, that gets attached to one person only, and doesn't get new targets afterwards because they spent most of the round as not a creep.

They both play differently because the roundstart creep will run out of time if they let their target live to long past their protect time (in total, you're going to need to protect for ~~45 minutes!) and the side antagonist creep gets time to set up the kill perfectly, they could get multiple creep objectives

ExcessiveUseOfCobblestone commented:
I'm not sure I like the idea of a person whose job it is to just kill people after a period of time on repeat. I do like the idea of a Twisted Nerve plot though where someone seems normal then they start acting stranger and stranger and the plot slowly evolves while the traitors are out stealing or whathaveyou.

If only being close to the person played that iconic theme....

tralezab commented:
alright, i can switch it to a side antag, but how can i make sure someone getting selected for a creep is targeting someone they have ties to? same department?

ExcessiveUseOfCobblestone commented:
It can stay random, fluff wise you see them pass by the hall (in an earlier shift) and something during the course of the round triggered your psychosis that turned a workplace crush to a desperate longing on it's way to a maddening obsession.

tralezab commented:
alrighty here i go, good suggestions.

ExcessiveUseOfCobblestone commented on code/modules/antagonists/creep/creep.dm:
```diff
@@ -0,0 +1,104 @@
+///CREEPY OBJECTIVES///
+
+/datum/objective/protect/timed //protect someone for a set amount of time. Then, it doesn't really matter what happens to them :3
+	var/timer = 9000 //15 minutes
```
`15 MINUTES`

tralezab committed:
a lot of things

creep is in love with their obsession, fixed suicide check, protect cares about suicides, randomized (kinda) protect times, creep mood effects, and the hug objective!

--- so this changed to a side antag so this commit does even more and less and idk i lost all track

Cyprex commented:
I'd let the creep deciding to murder their target happen randomly, or not happen at all if they're unlucky. Also, are these full antags that are allowed to murderbone?

plapatin commented:
real official theme song of this pull request:
https://www.youtube.com/watch?v=yRYFKcMa_Ek

tuypo1 commented:
Perhaps a possible objective might be flashing them.
But that might be TOO creepy.

Maybe either collect an object with their fingerprints or discover the string for their fingerprints.

But yeah definitely side antag only.

tralezab commented:
>Perhaps a possible objective might be flashing them.

i uhhhhhhhhh gotta hit you with a no on that one dawg

>Maybe either collect an object with their fingerprints or discover the string for their fingerprints.

this is an okay idea,  but you kinda get everything they have after you murder them. i've been keeping my objectives to things you can only get while they are alive.

tralezab commented:
![image](https://user-images.githubusercontent.com/40974010/50379749-d5ea8e00-0606-11e9-9f80-8e480cdf0d76.png)
I'll fix the objectives when I can, issues with assigning targets. But uh, high scores anyone?

tralezab commented:
I'm moving the guts of the antag quest checking and other stuff into a brain trauma that gives you the antag status which is SUPER great for doing what i want to do, the effects of the trauma is more persistent, and if you would lose the backend of the antag stuff you should lose the antag status itself. so fucking clean, hell yes.

wesoda25 commented:
Ok so on one hand this seems pretty cool but on the other don’t let cats or lizards or nonhumans in general be creeps or creep victims.

tralezab commented:
why?

tralezab committed:
a lot of fluff, features, and bugfixes. and bugs.

tralezab committed:
mood, admin adding, the event, objective picking, it all works now!

tralezab committed:
testing, and last minute fixes

tralezab commented:
everything works, i've tested it at least a fair bit. Really all I gotta do now is add more objectives and maybe the jeepers mode if i wanna bother

tralezab commented on code/modules/antagonists/creep/creep.dm:
```diff
@@ -0,0 +1,174 @@
+	return timer <= 0 || freeobj
+
+/*
+/datum/objective/hug
```
hey, any good way to make this work by the way?

Qustinnus replied:
Send a signal when a player is hugged

tralezab replied:
good idea

tralezab commented on code/datums/mood_events/generic_negative_events.dm:
```diff
@@ -137,11 +137,32 @@
+/datum/mood_event/notcreepingsevere//not hidden since it's so severe
+	description = "<span class='boldwarning'>OBSESSIONNNN WHERE ARE YOUUUUUUUUUUUUU?!</span>\n"
+	mood_change = -10
+	timeout = 100
```
roundend stat for collecting how long you've been close to the target, and if you are away too long your mood (visibly) tanking.

tralezab commented:
![image](https://user-images.githubusercontent.com/40974010/50391907-6f489b80-06fe-11e9-998e-e63b88eeea80.png)
Hug objective done, any other ideas that aren't flashing?

Qustinnus commented:
Get some of their hair, would have to be done with a do_after on the person.
Steal something valuable from said person (Check if the person has the heirloom trait maybe?)
Kill a co-worker (jealousy)

tralezab commented:
sweet and dammit those aren't that difficult to do. lets do this.

tralezab committed:
hugs

Qustinnus commented:
another one idk if you have it; steal their forensics. blood, fingerprints. whatever.

tralezab commented:
i'll stick with hair since it's more in theme and movie stalker-ish, also it's really easy to escape with someone's blood. i'm not sure how forensics and fingerprints work really

tuypo1 commented:
I am very much in favour of kill a co-worker objectives.

Maybe have a chance for it to be a random Borg instead of co worker as well (ai might be a bit much though)

MrDoomBringer commented:
oh damn, i really dig the "time spent near someone" thing. good shit

tralezab commented:
![image](https://user-images.githubusercontent.com/40974010/50405756-06990780-076e-11e9-8143-c0392d15be7c.png)
hey look i got the hud working

MrDoomBringer commented:
[Objective: hug someone while they're taking a shower](https://github.com/tgstation/tgstation/pull/41986)
unironically though i could see some EMERGENT GAMEYPLAY via ""accidently"" covering someone in blood or something in order to cajole them into showering.

Qustinnus commented:
yeah right helen drop the soap stein

RogueSteampunker commented:
This is unironically cool as hell, and my only gripe is that we don't have a Michael Myers outfit sprites in the game for this to really get spooky

tralezab committed:
fixes creep admin hud and tries to fix hug objective, adds heirloom objective

tralezab commented:
hey if this gets testmerged, you might want to reroll/remove the hug objective until i fix that

Qustinnus commented on code/datums/brain_damage/creepy_trauma.dm:
```diff
@@ -0,0 +1,112 @@
+	var/choked_up
+	GET_COMPONENT_FROM(mood, /datum/component/mood, owner)
+	if(mood)
+		switch(mood.sanity)
```
modafoka dont use a switch statement for one statement

tralezab replied:
🤤

tralezab committed:
hug finally fixed

tralezab committed:
heirloom examine for creeps, new objective

BIG DICK FEATURE BOYS

tralezab committed:
forgot to include this

tralezab committed:
jesus christ i'm dumb

optimumtact commented on code/datums/brain_damage/creepy_trauma.dm:
```diff
@@ -0,0 +1,117 @@
+	if(obsession.stat == DEAD) //killing them "cures" you! kinda! ish!
+		return
+	var/foundyou = FALSE
+	for(var/mob/living/L in range(7, owner))
```
really? just do a get_dist between get_turf(self) and the get_turf(obsession) first so you only do the view check if they're actually within 7 tiles.

 there's no reason to look at every thing in a 7 block range every time just to find one mob you already have a reference to to do a distance check

tralezab replied:
i'll do that, it's much better

tralezab replied:
And while i'm here I should really change that to view instead of range.

optimumtact commented on code/datums/brain_damage/creepy_trauma.dm:
```diff
@@ -0,0 +1,117 @@
+		total_time_creeping += 20
+		time_spent_away = 0
+		if(attachedcreepobj)
+			attachedcreepobj.timer -= 20 //mob subsystem ticks every 2 seconds(?), remove 20 deciseconds from the timer. sure, that makes sense.
```
what is the purpose of this

tralezab replied:
if they have the objective to be near the target while they are alive, it will set the trauma's attachedcreepobj to itself. I was already counting when they were near and what mood event to give if they were away long enough, it was easy to add in this as well

I should comment that though, it isn't clear

optimumtact commented on code/datums/brain_damage/creepy_trauma.dm:
```diff
@@ -0,0 +1,117 @@
+
+/datum/brain_trauma/special/creep/on_say(message)
+	var/foundyou = FALSE
+	for(var/mob/living/L in range(7, owner))
```
again, bad idea.
and better yet, just set a flag var on life and check it everywhere else so you only do one check every 2 life ticks.

It doesn't matter if they stutter a bit after their obsession has left their view range.

tralezab replied:
done much better

optimumtact commented on code/datums/brain_damage/creepy_trauma.dm:
```diff
@@ -0,0 +1,117 @@
+	var/chosen_victim
+	var/list/possible_targets = list()
+	var/list/viable_minds = list()
+	for(var/mob/Player in GLOB.mob_list)
```
There's a player list I believe, use it instead

optimumtact commented on code/modules/antagonists/creep/creep.dm:
```diff
@@ -0,0 +1,280 @@
+	if(target && target.current)
+		explanation_text = "Murder [target.name], the [!target_role_type ? target.assigned_role : target.special_role]."
+	else
+		explanation_text = "Free Objective"
```
Shouldn't this say something relevant?

tralezab replied:
Honestly it should be telling the admins something, like the trauma not finding an obsession but still getting objectives forged

tralezab replied:
done

optimumtact commented on code/modules/antagonists/creep/creep.dm:
```diff
@@ -0,0 +1,280 @@
+	var/timer = 3000 //5 minutes
+
+/datum/objective/spendtime/update_explanation_text()
+	if(timer == 1800)//just so admins can mess with it
```
what is the purpose of this?

optimumtact commented on code/modules/events/creep_awakening.dm:
```diff
@@ -0,0 +1,23 @@
+	fakeable = FALSE
+
+/datum/round_event/creep/start()
+	for(var/mob/living/carbon/human/H in shuffle(GLOB.alive_mob_list))
```
use the player list

tralezab committed:
optimumtact's review + no special/antag role creeps

tralezab commented on code/datums/brain_damage/creepy_trauma.dm:
```diff
@@ -0,0 +1,114 @@
+/datum/brain_trauma/special/creep/on_life()
+	if(!obsession)//gibbing, etc
+		return
+	if(get_dist(get_turf(owner), get_turf(obsession)) > 7 || obsession.stat == DEAD)
```
true true

tralezab committed:
tralezab optimumtact's review p2

tralezab committed:
forgot this over christmas!

tralezab committed:
WHOOPS

tralezab commented:
@Cyprex i suppose you could but you'd be really getting fucked with mood. i'll make a wiki page.

tralezab commented:
Wiki page done, I can just delete it if this doesn't get merged. It's not amazing but it'll do!

Cyprex commented:
Thanks for the wiki page!

optimumtact commented:
Can you make it clear in the notes and the announce to the user when they're selected that they are expected to follow all server rules, including rule 8

tralezab commented:
@optimumtact "<span class='boldannounce'>This role is not a license to break rules, ESPECIALLY RULE 8. Make stories and have fun with your role because it's a game we're all trying to enjoy, but know the limits. Don't do something a regular antagonist would get banned for.</span>"

sounds good? i'm not sure what you meant by notes, do you just mean a comment in the code, or in the memory? the wiki page?

@Cyprex I also have an icon for the creep and it's spot on the job page, but lets just wait for the actual merge.

tralezab committed:
creep warning message

Cyprex commented:
Aren't they solo antags?

> This role is not a license to break rules, ...

To me that implies all rules are in effect, and they do not have antag status.

tralezab commented:
Rule 4, lone antagonists can do whatever they want.

Cyprex commented:
Yes, the way it's worded makes it sound to the player like they do not have antag status.

MrDoomBringer commented:
@tralezab when oranges says notes he's referring to the player's memories (accessible via the View Notes verb)

@Cyprex this is refering to server rules, which antags and nonantags must both follow. [Read up on them here.](https://tgstation13.org/wiki/Rules)

Also, @tralezab, personally I would write something more clear like:

> **"The gods would like to remind you that this role, as with all other antags, does not allow you to break _ANY_ server rules, especially Rule 8 (These rules being listed from the "Rules" button at the top right of your mind's screen)**. Feel free to murder and pillage just like any other antag, though.

tralezab commented:
that works better, i'll go with that

tralezab committed:
MrDoomBringer's gmod heavyrp suggestion

tuypo1 commented:
>Feel free to murder and pillage just like any other antag, though.

Why?
This isn't a traitor role, it should not be a license to murderbone.

tralezab commented:
You're a delusional dude. You'd be hit with some pretty bad mood effects and you don't even get anything to murder bone with. Hell if you're caught by security they'll probably just remove your antag status. Stealth is definitely much more supported, and I don't want to discourage killing people to protect your target, throw them off, frame, things traitors do.

MrDoomBringer commented:
@tuypo1 you know that this is an antag role, right? and literally every /tg/ antag role allows murderboning?

tuypo1 commented:
Every other antag role random killing makes sense.

MrDoomBringer commented:
@tralezab also improve ur changelog my guy

MrDoomBringer commented on code/datums/brain_damage/creepy_trauma.dm:
```diff
@@ -0,0 +1,117 @@
+		if(!obsession)//we didn't find one
+			lose_text = ""
+			qdel(src)
+	owner.apply_status_effect(STATUS_EFFECT_INLOVE, obsession)
```
```suggestion
	gain_text = "<span class='warning'>You feel a strange attachment to [obsession].</span>"
	owner.apply_status_effect(STATUS_EFFECT_INLOVE, obsession)
```

tralezab replied:
I thought this would be annoying to do but it's really moving the ..() further down, sick

tralezab replied:
i did this, it actually looks pretty great

MrDoomBringer commented on code/datums/brain_damage/creepy_trauma.dm:
```diff
@@ -0,0 +1,117 @@
+	name = "Erotomania"
+	desc = "Patient has a subtype of delusional disorder, becoming irrationally attached to someone."
+	scan_desc = "severe erotomaniac delusions"
+	gain_text = ""
```
```suggestion
	gain_text = "<span class='warning'>You feel a strange attachment to someone.</span>"
```

tralezab replied:
done

MrDoomBringer commented on code/datums/brain_damage/creepy_trauma.dm:
```diff
@@ -0,0 +1,117 @@
+			shake_camera(owner, 15, 1)
+		if(4)
+			to_chat(owner, "<span class='warning'>You faint.</span>")
+			owner.Unconscious(80)
```
>determine someone is creeping on you
>cuff them and lock them in a room with you so they die of fucking spaghetti overdose

tralezab replied:
the fact people are already coming up with stories like this makes me pretty happy

Qustinnus replied:
do not dare cuff me @MrDoomBringer

MrDoomBringer replied:
uwu

tralezab commented:
do not merge for a second, i need to fix a runtime.

tralezab commented on code/datums/status_effects/neutral.dm:
```diff
@@ -68,4 +68,4 @@
 /datum/status_effect/in_love/tick()
 	if(date)
-		new /obj/effect/temp_visual/love_heart/invisible(get_turf(date.loc), owner)
+		new /obj/effect/temp_visual/love_heart/invisible(get_turf(date.loc), owner)
```
really helpful, thank you!

tralezab committed:
It's easy to get taken in by the spectacle, to enjoy a bit of naughty amusement. But make no mistake: the Creep is a danger.

tralezab committed:
Merge branch 'master' into two-faced-creep

Qustinnus commented:
@tralezab I just had a random idea; 2 people creeping on the same person and thus having to kill each other, but they dont actually know who the other creep is and have to find out.

Qustinnus commented:
probably better to do in a future PR to prevent feature creep tho

tralezab commented:
hahahahahaha ok sweet dude

yeah i just want this done i've been working on it since 2018

Qustinnus commented:
you make that sound like a long time but its january first so everyone has been working on their pr since 2018

MrDoomBringer commented:
good shit @tralezab , based pr

ExcessiveUseOfCobblestone commented:
i'm so glad underwear isn't a clothing item after this.

Cyprex commented:
Wubli reported having a Murder and Assassinate objective for the same person, is this intended?

tralezab commented:
@Cyprex The creep has two assassinate objectives, but their explanation says you must "Murder" the target. I'd have to assume they were a traitor or some other antagonist with an assassinate objective.

ExcessiveUseOfCobblestone commented:
Put it on the tracker if it's an actual bug since this is merged now.

bobbahbrown commented:
nobody noticed this pr broke picture datum

kind of cringe

ExcessiveUseOfCobblestone commented:
lol
